tests/lib/test_phase_stats.py

```
@@ -895,6 +895,117 @@ def test_calculate_co2_intensity_uses_microgram_scale():
+    assert by_metric['psu_energy_cgroup_slice']['type'] == 'TOTAL'
+    assert by_metric['psu_power_cgroup_container']['type'] == 'MEAN'
+    assert by_metric['psu_energy_cgroup_container']['type'] == 'TOTAL'
+
+    machine_power_baseline_mw = Decimal(baseline_energy_uj) / baseline_duration_us * Decimal(1000)
+    machine_power_current_phase_mw = Decimal(sub1_energy_uj) / sub1_duration_us * Decimal(1000)
+
+    surplus_power_expected = machine_power_current_phase_mw - machine_power_baseline_mw
+    # baseline energy over the phase = power_mW * duration_us * 1e-3
+    # (mW * duration_s = mJ; duration_s = duration_us * 1e-6; mJ * 1000 = uJ -> combined factor 1e-3),
+    # matching production's Decimal(1e-3). round() mirrors generate_csv_line()'s own rounding on store.
+    surplus_energy_expected = round(Decimal(sub1_energy_uj) - (machine_power_baseline_mw * sub1_duration_us * Decimal(1e-3)))
```

Ruff (0.16.1)

[warning] `Decimal()` called with float literal argument

Replace with string literal

(RUF032)